tf2/src/buffer_core.cpp

```diff
@@ -1666,31 +1669,31 @@ void BufferCore::_chainAsVector(const std::string & target_frame, TimePoint targ
         assert(0);
       }
     }
-    int m = (int)target_frame_chain.size()-1;
-    int n = (int)source_frame_chain.size()-1;
-    for (; m >= 0 && n >= 0; --m, --n)
+    size_t tfs = target_frame_chain.size() - 1;
+    size_t sfs = source_frame_chain.size() - 1;
+    size_t m = 1;
+    size_t n = 1;
+    for (; m <= tfs && n <= sfs; ++m, ++n)
     {
-      if (source_frame_chain[n] != target_frame_chain[m])
+      if (source_frame_chain[sfs - n] != target_frame_chain[tfs - m])
         break;
     }
     // Erase all duplicate items from frame_chain
-    if (n > 0)
-      source_frame_chain.erase(source_frame_chain.begin() + (n-1), source_frame_chain.end());
+    if (n <= sfs) {
```

Comment:
@ivanpauno I think it should be 
```suggestion
    if (n < sfs) {
```
instead.

Reply:
Yeap, fixed